microsoft/botframework-solutions
# [C#][VA/Skill] Integrate specific cultures

**This Pull Request will be present in the next release.**

## Purpose

_What is the context of this pull request? Why is it being done?_

We updated the _Virtual Assistant_ and the _Skill_ language to replace the language and start to use complete cultures for each language with the objective of improve the functionality of the **multilanguage** implementation. 

## Changes
_Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)_
- Implement `fallback to root language` functionality
- Rename different folders and files with each _specific culture_ name
- Use of the _specific culture_ name for the _deployment scripts_ files
- Changes in _mainDialog_ implementation to obtain the new locale language cultures (eg. 'en-us').
- Update template file to include the new changes when generate a new _VA_ and _Skill_ samples.

## Tests
_Is this covered by existing tests or new ones? If no, why not?_

It's covered by _LocalizationTests_, _MainDialogTests_ files.

## Testing steps

1. Deploy a _Virtual Assistant_ 
2. Verify in the bot solution that the resource folder contains the specific cultures
3. Execute the _Virtual Assistant_ 
4. Check the behavior of the _Virtual Assistant_ communication

## Feature Plan
_Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues._
*-*

### Checklist

#### General

#### Bots

#### Deployment Scripts
